integration/helpers/client_provider.py

```diff
@@ -73,5 +96,106 @@ def sfn_client(self):
         Step Functions Client
         """
         if not self._sfn_client:
-            self._sfn_client = boto3.client("stepfunctions")
+            with self._lock:
+                if not self._sfn_client:
+                    self._sfn_client = boto3.client("stepfunctions")
         return self._sfn_client
+
+    @property
+    def cloudwatch_log_client(self):
+        """
+        CloudWatch Log Client
+        """
+        if not self._cloudwatch_log_client:
+            with self._lock:
+                if not self._cloudwatch_log_client:
```

Comment:
Why `if not self._cloudwatch_log_client:` appear again? Same for other places.

Reply:
Double lock checking, so that we don't acquire the lock if the value is already instantiated (mainly for performance).
We may not have performance requirements that high though, so maybe we can drop the first `if` for readability.

Reply:
I think you can safely remove the first `if`

Reply:
Got it, removed the first if.